core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreAggregateCaseAggregations.java

```diff
@@ -363,6 +367,14 @@ private Optional<CaseAggregation> extractCaseAggregation(Symbol aggregationSymbo
             return Optional.empty();
         }
 
+        Expression whenOperand = caseExpression.getWhenClauses().getFirst().getOperand();
```

Comment:
Move this whole section closed to where it's used -- i.e., right above `return Optional.of(new CaseAggregation(...)`

Reply:
Moved closer